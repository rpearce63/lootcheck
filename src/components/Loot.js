import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchBitcoin } from "../actions/bitcoin";

export class Loot extends Component {
  componentDidMount() {
    this.props.fetchBitcoin();
  }

  computeBitcoin() {
    const bitcoinRate = this.getRate();
    if (!bitcoinRate) return "";
    return this.props.balance / parseInt(bitcoinRate.replace(",", ""), 10);
  }

  getRate() {
    const { bitcoin } = this.props;
    if (Object.keys(bitcoin).length === 0) return "";
    return bitcoin.bpi.USD.rate;
  }
  render() {
    return (
      <div>
        <h3>Bitcoin balance: {this.computeBitcoin()}</h3>
        <small className="bitcoin-rate">Bitcoin value: {this.getRate()}</small>
      </div>
    );
  }
}

export default connect(
  state => state,
  { fetchBitcoin }
)(Loot);
