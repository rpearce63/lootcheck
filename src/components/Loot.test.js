import React from "react";
import { shallow, mount } from "enzyme";
import { Loot } from "./Loot";

describe("Loot", () => {
  let props = { balance: 10, bitcoin: {} };
  let loot = shallow(<Loot {...props} />);

  it("renders properly", () => {
    expect(loot).toMatchSnapshot();
  });

  describe("when mounted", () => {
    const mockFetchBitcoin = jest.fn();
    beforeEach(() => {
      props.fetchBitcoin = mockFetchBitcoin;
      loot = mount(<Loot {...props} />);
    });

    it("dispatches the `fetchBitcoin()` method it receives from props", () => {
      expect(mockFetchBitcoin).toHaveBeenCalled();
    });
  });

  describe("when there are valid bitcoin props", () => {
    beforeEach(() => {
      props = { balance: 10, bitcoin: { bpi: { USD: { rate: "1,000" } } } };
      loot = shallow(<Loot {...props} />);
    });

    it("displays the correct bitcoin value", () => {
      expect(loot.find("h3").text()).toEqual("Bitcoin balance: 0.01");
    });

    it("displays the current bitcoin rate", () => {
      expect(loot.find(".bitcoin-rate").text()).toEqual("Bitcoin value: 1,000");
    });
  });
});
