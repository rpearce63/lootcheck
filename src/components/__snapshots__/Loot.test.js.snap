// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loot
    balance={10}
    bitcoin={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Bitcoin balance: 
          
        </h3>,
        <small
          className="bitcoin-rate"
        >
          Bitcoin value: 
          
        </small>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Bitcoin balance: ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Bitcoin balance: ",
          "",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Bitcoin value: ",
            "",
          ],
          "className": "bitcoin-rate",
        },
        "ref": null,
        "rendered": Array [
          "Bitcoin value: ",
          "",
        ],
        "type": "small",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Bitcoin balance: 
            
          </h3>,
          <small
            className="bitcoin-rate"
          >
            Bitcoin value: 
            
          </small>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Bitcoin balance: ",
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Bitcoin balance: ",
            "",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Bitcoin value: ",
              "",
            ],
            "className": "bitcoin-rate",
          },
          "ref": null,
          "rendered": Array [
            "Bitcoin value: ",
            "",
          ],
          "type": "small",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
